### PR TITLE
fix: fixed a testing case for brave browser

### PR DESCRIPTION
**Pull request by rebelchris**
In chrome we use the default new tab as: `chrome-search://local-ntp/local-ntp.html`.
Brave however doesn't provide a inner tab url, so we have to resolve to setting the chrome search default like so: `chrome://new-tab-page`.

I've added a constant check for brave browser.
Running it in Chrome:
Screenshot 2021-12-06 at 11 34 41

And in Brave:
Screenshot 2021-12-06 at 11 34 35

### CODE REVIEW

File: packages/extension/src/newtab/dnd.ts [hunk 1]
```diff
@@ -15,9 +16,14 @@ export enum CustomTime {
...
 const DEFAULT_URL = 'https://www.google.com';
 const CHROME_DEFAULT_URL = 'chrome-search://local-ntp/local-ntp.html';
+const BRAVE_DEFAULT_URL = 'chrome://new-tab-page';
 
-export const getDefaultLink = (): string =>
-  process.env.TARGET_BROWSER === 'chrome' ? CHROME_DEFAULT_URL : DEFAULT_URL;
+export const getDefaultLink = (): string => {
+  const browserTest = () =>
+    process.env.TARGET_BROWSER === 'chrome' ? CHROME_DEFAULT_URL : DEFAULT_URL;
+  const braveTest = () => (isBrave() ? BRAVE_DEFAULT_URL : browserTest());
```
Comment (idoshamun) [inline packages/extension/src/newtab/dnd.ts:24]: Looks like we can take this out of the function.
Functions generating more functions inside usually utilize the parent function's scope but here it's not necessary :)

File: packages/shared/src/lib/constants.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// We have to declare the navigator because brave is not set in the types of the navigator, yet it does exist
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const navigator: any;
```
Comment (idoshamun) [inline packages/shared/src/lib/constants.ts:3]: How about this?

```suggestion
declare const navigator: Navigator & {brave?: {isBrave: unknown}};
```

File: packages/shared/src/lib/constants.ts [hunk 3]
```diff
@@ -15,3 +18,16 @@ export const isDevelopment = process.env.NODE_ENV === 'development';
...
+  if (!window.Promise) {
+    return false;
+  }
+  if (
+    typeof navigator.brave === 'undefined' ||
+    typeof navigator.brave.isBrave !== 'function'
+  ) {
+    return false;
+  }
+  return true;
```
Comment (idoshamun) [inline packages/shared/src/lib/constants.ts:32]: I think utilizing the optional chaining and returning the result of the statement is the way to go :)

```suggestion
  return typeof navigator.brave?.isBrave === 'function';
```

**Review by idoshamun — CHANGES_REQUESTED** [review]
I have some ideas for refactoring. Other than that looks great. Quick response time :)

**Comment by rebelchris** [comment]
@idoshamun Thanks! Refactored with your comments.
Missed a couple easy wins there 👀

File: packages/extension/src/newtab/dnd.ts [hunk 4]
```diff
@@ -18,12 +18,11 @@ const DEFAULT_URL = 'https://www.google.com';
...
-  const browserTest = () =>
-    process.env.TARGET_BROWSER === 'chrome' ? CHROME_DEFAULT_URL : DEFAULT_URL;
-  const braveTest = () => (isBrave() ? BRAVE_DEFAULT_URL : browserTest());
-  return braveTest();
-};
+const browserTest = () =>
+  process.env.TARGET_BROWSER === 'chrome' ? CHROME_DEFAULT_URL : DEFAULT_URL;
+const braveTest = () => (isBrave() ? BRAVE_DEFAULT_URL : browserTest());
+
+export const getDefaultLink = (): string => braveTest();
```
Comment (idoshamun) [inline packages/extension/src/newtab/dnd.ts:25]: Another easy win 👀 

```suggestion
const getDefaultLink = () => (isBrave() ? BRAVE_DEFAULT_URL : browserTest());
```

**Review by idoshamun — APPROVED** [review]
I approved it but just make sure to apply this one tiny change

**Pull request merged**